test/athens/common_events/block_test.clj

```diff
@@ -778,7 +779,288 @@
+          target-parent-str "a"
+          target-uid        "test-target-uid"
+          target-text       "b"
+          source-uid        "test-source-uid"
+          source-text       "c"
+          setup-txs         [{:db/id          -1
+                              :node/title     "test page"
+                              :block/uid      "page-uid"
+                              :block/children [{:db/id          -2
+                                                :block/uid      target-parent-uid
+                                                :block/string   target-parent-str
+                                                :block/order    0
+                                                :block/children [{:db/id          -3
+                                                                  :block/uid      target-uid
+                                                                  :block/string   target-text
+                                                                  :block/order    0
+                                                                  :block/children []}
+                                                                 {:db/id          -4
+                                                                  :block/uid      source-uid
+                                                                  :block/string   source-text
+                                                                  :block/order    1
+                                                                  :block/children []}]}]}]]
+
+
+      (d/transact @fixture/connection setup-txs)
+      (let [source-block            (common-db/get-block @@fixture/connection [:block/uid source-uid])
+            target-block            (common-db/get-block @@fixture/connection [:block/uid target-uid])
+            target-parent-block     (common-db/get-block @@fixture/connection [:block/uid target-parent-uid])
+            drop-same-parent-event  (common-events/build-drop-same-event -1
+                                                                         :above
+                                                                         source-uid
+                                                                         target-uid)
+            drop-same-parent-txs    (resolver/resolve-event-to-tx @@fixture/connection drop-same-parent-event)]
+        (t/is (= 2 (-> target-parent-block :block/children count)))
+        (t/is (= 0 (:block/order target-block)))
+        (t/is (= 1 (:block/order source-block)))
+
+        (d/transact @fixture/connection drop-same-parent-txs)
+        (let [source-block         (common-db/get-block @@fixture/connection [:block/uid source-uid])
+              target-block         (common-db/get-block @@fixture/connection [:block/uid target-uid])
+              target-parent-block  (common-db/get-block @@fixture/connection [:block/uid target-parent-uid])]
+          (t/is (= 2 (-> target-parent-block :block/children count)))
```

Comment:
@neotyk  This test is failing giving children as 4 instead of 2

Reply:
and what if you try `get-block` with `:db/id`?

Reply:
We've reported this bug to Datahike, @filipesilva has details.